### DISCUSSION TITLE
Python

### DISCUSSION THREAD

**Discussion by Keviniscaiji** [community/community#32986] [Other Feature Feedback, Questions, & Ideas] [answered]
why public key cryptography can support the authentication of the origin of a message.

**Comment by WangHewei16** [comment, marked as answer]
Hello! Maybe it is a security question. The entire public key cryptography process requires the cooperation of the sender's public and private keys. The sender encrypts with its own private key and sends its own public key to the receiver. Using the public key, the receiver can decrypt the message. Because the sender's private key is kept secret, if a middleman forges the encrypted content, it cannot be restored with the public key to determine the authentication of the message source.

**Reply by Keviniscaiji** [reply]
Thanks, it is really useful!
